### PR TITLE
Adding visualization - tensorboard, accuracy-score, confusion-matrix for Cifar 10 Example

### PR DESCRIPTION
**Pull request by shrinath-suresh**
Signed-off-by: Shrinath Suresh <shrinath@ideas2it.com>

### CODE REVIEW

File: cifar10/utils.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+                data.append((vocab[target_index], vocab[predicted_index], count))
+
+        confusion_matrix_df = pd.DataFrame(data, columns=["target", "predicted", "count"])
+
+        confusion_matrix_key = os.path.join(folder_name, "confusion_df.csv")
+
+        # Logging confusion matrix to ss3
+        csv_buffer = StringIO()
+        confusion_matrix_df.to_csv(csv_buffer, index=False, header=False)
+        s3_resource = boto3.resource("s3")
```
Comment (chauhang) [inline cifar10/utils.py:93]: @shrinath-suresh Can we remove dependency on boto / s3 and use pvc/volumes instead?

File: cifar10/pipeline.py [hunk 2]
```diff
@@ -80,13 +80,13 @@ def ls(input_dir: str):
             "/pvc/input/cifar10_train.py",
         ],
         output_data = ["/pvc/output/train/models"],
-        input_parameters = [{"tensorboard_root": "/pvc/output/train/tensorboard", 
+        input_parameters = [{"tensorboard_root": "s3://kubeflow-dataset/tensorboardX",
```
Comment (chauhang) [inline cifar10/pipeline.py:83]: @shrinath-suresh Please remove the dependency on S3

File: cifar10/pipeline.py [hunk 3]
```diff
@@ -80,13 +80,13 @@ def ls(input_dir: str):
...
         "max_epochs": 1, "gpus": 0, "train_batch_size": None, "val_batch_size": None, "train_num_workers": 4, 
         "val_num_workers": 4 , "learning_rate": 0.001, 
-        "accelerator": None}],
-        source_code = ["https://kubeflow-dataset.s3.us-east-2.amazonaws.com/cifar10_datamodule.py", "https://kubeflow-dataset.s3.us-east-2.amazonaws.com/cifar10_train.py"],
+        "accelerator": None, "bucket_name": "kubeflow-dataset", "folder_name": "Cifar10Viz"}],
+        source_code = ["https://kubeflow-dataset.s3.us-east-2.amazonaws.com/cifar10_datamodule.py", "https://kubeflow-dataset.s3.us-east-2.amazonaws.com/cifar10_train.py", "https://kubeflow-dataset.s3.us-east-2.amazonaws.com/utils.py"],
```
Comment (chauhang) [inline cifar10/pipeline.py:87]: @shrinath-suresh Please use the git url instead, remove dependency on s3

**Review by chauhang — CHANGES_REQUESTED** [review]
Please see comments inline for s3 dependencies in the code

**Comment by shrinath-suresh** [comment]
Duplicate. Merged using the following PRs https://github.com/chauhang/pytorch-pipeline/pull/19 and https://github.com/chauhang/pytorch-pipeline/pull/17
